ui/src/menu.tsx

```diff
@@ -38,9 +38,9 @@ export interface Menu {
   name?: Id
 }
 
-export const XMenu = ({ model }: { model: Menu }) => {
+export const XMenu = ({ model }: { model: Menu & {hideCaret?: B } }) => {
```

Comment:
I'm not in favor of reusing this component for the table cell menu - the `& {hideCaret?: B }` breaks pattern, and the styling customizations done here may not be relevant to the table cell menu. It should be fairly simply to code up a menu in the cell_type tsx file.